packages/dd-trace/src/log.js

```diff
@@ -43,44 +38,72 @@ const memoize = func => {
   return memoized
 }
 
+function processArgs (args) {
+  if (typeof args[0] === 'function') {
+    args = args[0]()
+    return Array.isArray(args) ? args : [args]
+  }
+  return args
+}
+
 const log = {
+  _isLogLevelEnabled (level) {
+    return _logLevels[level] >= this._logLevel
+  },
+
   use (logger) {
     if (logger && logger.debug instanceof Function && logger.error instanceof Function) {
-      _logger = logger
+      this._logger = logger
     }
 
     return this
   },
 
   toggle (enabled, logLevel) {
-    _enabled = enabled
-    _logLevel = _checkLogLevel(logLevel)
+    this._enabled = enabled
+    this._logLevel = _checkLogLevel(logLevel)
 
     return this
   },
 
   reset () {
-    _logger = _default
-    _enabled = false
-    _deprecate = memoize((code, message) => {
-      _logger.error(message)
+    this._logger = _default
+    this._enabled = false
+    this._deprecate = memoize((code, message) => {
+      this._logger.error(message)
       return this
     })
-    _logLevel = _checkLogLevel()
+    this._logLevel = _checkLogLevel()
+
+    return this
+  },
+
+  debug (...message) {
+    if (this._enabled && this._isLogLevelEnabled('debug')) {
+      this._logger.debug(...processArgs(message))
+    }
+
+    return this
+  },
+
+  info (...message) {
+    if (this._enabled && this._isLogLevelEnabled('info')) {
+      this._logger.info(...processArgs(message))
+    }
 
     return this
   },
 
-  debug (message) {
-    if (_enabled && _isLogLevelEnabled('debug')) {
-      _logger.debug(message instanceof Function ? message() : message)
+  warn (...message) {
```

Comment:
The most popular logging libraries are supported out of the box with the current approach. This approach will only work well with `console.log` and force a breaking change on all the others to add logic to split the arguments as individual logs.

Reply:
Fixed in latest rev